includes/api/order.php

```diff
@@ -119,8 +119,31 @@ function createWcOrder(WP_REST_Request $request)
     }
 
     $order = wc_get_order($orderId);
-
-    if ($order) {
+    
+    if($order){
+
+        $disableCouponFlag = false;
+
+        // Woo dynamic discount price plugin
+        if(is_plugin_active('yith-woocommerce-dynamic-pricing-and-discounts-premium/init.php')) {
+
+            foreach ($order->get_items() as $itemId => $item) {
+                $dynamicRules = $item->get_meta('_ywdpd_discounts');
+
+                foreach ($dynamicRules['applied_discounts'] as $appliedDiscount) {
+                    if (isset( $appliedDiscount['set_id'])){
+                        $ruleId = $appliedDiscount['set_id'];
+                        $rule    = ywdpd_get_rule($ruleId);
```

Comment:
@poojarazor Do we need to initiate a class? to invoke this function?

Reply:
it's a public function , no need to initiate any class